server/src/api/comment.rs

```diff
@@ -474,11 +474,11 @@ impl Perform for Oper<RemoveComment> {
     }
 
     // Verify that only a mod or admin can remove
-    let mods_and_admins = blocking(pool, move |conn| {
-      Community::community_mods_and_admins(conn, community_id)
+    let is_mod_or_admin = blocking(pool, move |conn| {
+      Community::is_mod_or_admin(conn, user_id, community_id)
     })
-    .await??;
-    if !mods_and_admins.contains(&user_id) {
+    .await?;
+    if !is_mod_or_admin {
       return Err(APIError::err("not_an_admin").into());
     }
```

Comment:
This isnt what I meant, I was talking about a function that also handles the if condition. So all you would need to put here is `is_mod_or_admin(user_id)?`. If its a mod or admin, it does nothing, otherwise it returns an error.

Reply:
I'm not sure what you mean. I can't use a `?`, I have to match no matter what, so I can return the APIError

Reply:
This is what I mean:
```
async fn is_mod_or_admin(pool: &DbPool, user_id: i32, community_id: i32) -> Result<(), LemmyError> {
  let is_mod_or_admin = blocking(pool, move |conn| {
    Community::is_mod_or_admin(conn, user_id, community_id)
  })
    .await?;
  if !is_mod_or_admin {
    return Err(APIError::err("not_an_admin").into());
  }
  return Ok(())
}
```

Then for the actual check in an API call, you just need to do `is_mod_or_admin(pool, user_id, community_id).await?;`.

Reply:
Okay I added helpers like this for `is_mod_or_admin`, and another for `is_admin`